### PR TITLE
Pv2 initial approvals integration pv2 1019 pv2 1020 pv2 1021 pv2 1190

### PR DESCRIPTION
**Pull request by abroodie**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+            var sanitization = transportConfig.Sanitization();
+            var strategy = sanitization.UseStrategy<ValidateAndHashIfNeeded>();
+            strategy.RuleNameSanitization(
+                ruleNameSanitizer: ruleName => ruleName.Split('.').LastOrDefault() ?? ruleName);
+            endpointConfig.UseSerialization<NewtonsoftSerializer>();
+            endpointConfig.EnableInstallers();
+        }
+
+        //[BeforeTestRun(Order = 52)]
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: can this be removed?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Approvals Interface/PV2-688 - New Apprenticeship Approved.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿Feature: PV2-688 - New Apprenticeship Approved
+	As an Employer
+	I want tthe Payments Calc Data-Locks process to be made aware of newly approved commitments
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Approvals Interface/PV2-688 - New Apprenticeship Approved.feature]: typo "tthe"

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/ApprovalsService.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            };
+        }
+
+        /// <summary>
+        /// This is the main entry point for your service instance.
+        /// </summary>
+        /// <param name="cancellationToken">Canceled when Service Fabric needs to shut down this service instance.</param>
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            // TODO: Replace the following sample code with your own logic
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/ApprovalsService.cs]: can you remove this

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        /// <returns></returns>
+        public async Task<string> OpenAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                logger.LogDebug($"Opening endpoint: {config.EndpointName}");
+                if (endpointInstance == null)
+                {
+                    var endpointConfiguration = CreateEndpointConfiguration();
+                    //endpointConfiguration.UseContainer<AutofacBuilder>(c => c.ExistingLifetimeScope(lifetimeScope));
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: needed?

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            var persistence = endpointConfiguration.UsePersistence<AzureStoragePersistence>();
+            persistence.ConnectionString(config.StorageConnectionString);
+
+            endpointConfiguration.DisableFeature<TimeoutManager>();
+            var transport = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
+            transport
+                .ConnectionString(configHelper.GetConnectionString("DASServiceBusConnectionString"))
+                .Transactions(TransportTransactionMode.ReceiveOnly)
+                .RuleNameShortener(ruleName => ruleName.Split('.').LastOrDefault() ?? ruleName);
+            //            EndpointConfigurationEvents
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: Is this going to be used?

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            //                .OnConfiguringTransport(transport); //TODO: find AutoFac & NSB way to do this
+            endpointConfiguration.SendFailedMessagesTo(config.FailedMessagesQueue);
+            endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+            endpointConfiguration.EnableInstallers();
+            //          endpointConfiguration.Pipeline.Register(typeof(TelemetryHandlerBehaviour),
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: is this needed?

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/Ioc/Modules/DasMessagingModule.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+using SFA.DAS.Payments.Messages.Core;
+using SFA.DAS.Payments.ServiceFabric.Core;
+
+namespace SFA.DAS.Payments.DataLocks.ApprovalsService.Infrastructure.Ioc.Modules
+{
+    public class DasMessagingModule : Module
+    {
+        protected override void Load(ContainerBuilder builder)
+        {
+            //builder.Register((c, p) =>
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/Ioc/Modules/DasMessagingModule.cs]: is this needed?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs (hunk 1 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/ApprovalsInterfaceSteps.cs]: done

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Approvals Interface/PV2-688 - New Apprenticeship Approved.feature (hunk 2 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Tests/Approvals Interface/PV2-688 - New Apprenticeship Approved.feature]: done

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/ApprovalsService.cs (hunk 3 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.DataLocks.ApprovalsService/ApprovalsService.cs]: done

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs (hunk 4 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: done

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs (hunk 5 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: removed

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs (hunk 6 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/DasStatelessEndpointCommunicationListener.cs]: nope, removed

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/Ioc/Modules/DasMessagingModule.cs (hunk 7 above)
Comment (abroodie) [reply src/SFA.DAS.Payments.DataLocks.ApprovalsService/Infrastructure/Ioc/Modules/DasMessagingModule.cs]: nope, removed

**Pull request merged**